tests/components/wake_on_lan/test_switch.py

```diff
@@ -1,266 +1,287 @@
 """The tests for the wake on lan switch platform."""
-import unittest
+import platform
+import subprocess
+
+import pytest
+import wakeonlan
 
 import homeassistant.components.switch as switch
-from homeassistant.const import STATE_OFF, STATE_ON
-from homeassistant.setup import setup_component
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.setup import async_setup_component
 
 from tests.async_mock import Mock, patch
-from tests.common import get_test_home_assistant, mock_service
-from tests.components.switch import common
-
-TEST_STATE = None
+from tests.common import async_mock_service
 
 
-def call(cmd, stdout, stderr):
-    """Return fake subprocess return codes."""
-    if cmd[5] == "validhostname" and TEST_STATE:
-        return 0
-    return 2
+@pytest.fixture
```

Comment:
Ah snap, forgot something
```suggestion
@pytest.fixture(autouse=True)
```